crates/symbolicator-js/src/symbolication.rs

```diff
@@ -105,21 +105,29 @@ async fn symbolicate_js_frame(
     // we check for a valid line (i.e. >= 1) first, as we want to avoid resolving / scraping the minified
     // file in that case. we frequently saw 0 line/col values in combination with non-js files,
     // and we want to avoid scraping a bunch of html files in that case.
-    let line = if raw_frame.lineno > 0 {
-        raw_frame.lineno
-    } else {
-        return Err(JsModuleErrorKind::InvalidLocation {
-            line: raw_frame.lineno,
-            col: raw_frame.colno,
-        });
+    let line = match raw_frame.lineno {
+        Some(lineno) if lineno > 0 => lineno,
+        lineno => {
+            return Err(JsModuleErrorKind::InvalidLocation {
+                line: lineno.unwrap_or(0),
+                col: raw_frame.colno,
+            });
+        }
+    };
+
+    let abs_path = match raw_frame.abs_path {
+        Some(ref abs_path) => abs_path,
+        None => {
+            return Err(JsModuleErrorKind::InvalidAbsPath);
```

Comment:
I think we need to move this above the line check, the error closer to what we expect and when we have no `abs_path` we also will never have a `line`, so this error never fires.

Reply:
Agreed. A missing `abs_path` is a more fundamental error than a missing/invalid line number.

Also, this match can be written a bit more concisely:
```suggestion
    let abs_path = raw_frame
        .abs_path
        .as_ref()
        .ok_or(JsModuleErrorKind::InvalidAbsPath)?;

```

Reply:
Moved up in 1ff052e